TestsUploadUsers.cs

```diff
@@ -0,0 +1,206 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using NUnit.Framework;
+using RestSharp;
+
+namespace APIAutomation.Tests
+{
+    [TestFixture]
+    public class UploadUsersTests
+    {
+        private RestClient _clientForReadScope;
+        private RestRequest _requestForReadScopeGetUsers;
+        private RestRequest _requestForReadScopeGetZipCodes;
+
+        private RestClient _clientForWriteScope;
+        private RestRequest _requestForWriteScope;
+
+        [SetUp]
+        public void Setup()
+        {
+            var clientForReadScope = ClientForReadScope.GetInstance();
+            _clientForReadScope = clientForReadScope.GetRestClient();
+            _requestForReadScopeGetZipCodes = new RestRequest("/zip-codes");
+            _requestForReadScopeGetUsers = new RestRequest("/users");
+
+            var clientForWriteScope = ClientForWriteScope.GetInstance();
+            _clientForWriteScope = clientForWriteScope.GetRestClient();
+            _requestForWriteScope = new RestRequest("/users/upload", Method.Post);
+        }
+
+        [Test]
+        public void CheckAllUsersAreReplacedWithFiled_Test()
+        {
+            _clientForReadScope.AddDefaultHeader("Accept", "application/json");
+            RestResponse getAllUsersResponse = _clientForReadScope.Execute(_requestForReadScopeGetUsers);
+            List<User> initialUsers = JsonConvert.DeserializeObject<List<User>>(getAllUsersResponse.Content);
+
+            RestResponse getZipCodesResponse = _clientForReadScope.Execute(_requestForReadScopeGetZipCodes);
+            List<string> availableZipCodes = JsonConvert.DeserializeObject<List<string>>(getZipCodesResponse.Content);
+
+            List<User> users = new List<User>
+             {
+                 new User{Age = RandomUserGenerator.GenerateRandomAge(), Name = RandomUserGenerator.GenerateRandomName(), Sex = RandomUserGenerator.GenerateRandomSex(), ZipCode = availableZipCodes[0]},
+                 new User{Age = RandomUserGenerator.GenerateRandomAge(), Name = RandomUserGenerator.GenerateRandomName(), Sex = RandomUserGenerator.GenerateRandomSex(), ZipCode = availableZipCodes[1]}
+             };
+            var usersCountFile = users.Count;
+
+            string jsonFilePath = "C:\\Users\\kosac_xmbzi23\\OneDrive\\Рабочий стол\\API_Automation_CSharp\\API_C_sharp\\Test_data\\Users.json";
```

Comment:
Please, don't use absolute paths in your code, better implement [relative](https://stackoverflow.com/questions/40994534/get-relative-path-of-a-file-c-sharp) file access.

Reply:
ok, it is done